SignalMessaging/attachments/ContactFieldView.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+public class ContactFieldView: UIView {
+
+    @available(*, unavailable, message: "use other constructor instead.")
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("Unimplemented")
+    }
+
+    public required init(rows: [UIView], hMargin: CGFloat) {
+        super.init(frame: CGRect.zero)
+
+        self.layoutMargins = .zero
+        self.preservesSuperviewLayoutMargins = false
+
+        addRows(rows: rows, hMargin: hMargin)
+    }
+
+    private func addRows(rows: [UIView], hMargin: CGFloat) {
+
+        var lastRow: UIView?
+
+        let addSpacerRow = {
+            guard let prevRow = lastRow else {
+                owsFail("\(self.logTag) missing last row")
+                return
+            }
+            let row = UIView()
+            row.backgroundColor = UIColor(rgbHex: 0xdedee1)
+            self.addSubview(row)
+            row.autoSetDimension(.height, toSize: 1)
+            row.autoPinLeadingToSuperviewMargin(withInset: hMargin)
```

Comment:
Something seems a little strange about this inset/margin juggling.

As I understand it:

- You've explicitly set the superview's horizontal margin to 0
- You then pin to that margin (which is now zero, so you could just as well pin to the edge here)
- Then you offset that margin pinning by an inset.

Since you've repeatedly zero'd out horizontal margins on all superviews this works out in the end, but it feels like you're fighting the paradigm. Rather than zeroing out the margin and then pinning to the margin, just pin to the edge. Or if you want a consistent offset, define that as your superviews margin and pin to that.

Apologies if I've covered this same code in another PR, but I keep seeing similar code.

Reply:
This has all be reworked in a subsequent PR.